ground/gcs/src/plugins/config/configgadgetwidget.cpp

```diff
@@ -260,12 +277,14 @@ void ConfigGadgetWidget::onAutopilotConnect() {
             QLabel *txt = new QLabel(this);
             txt->setText(tr("Board detected, but of unknown type. This could be because either your GCS or firmware is out of date."));
             qwd = txt;
-        }
-        else {
+        } else {
             qwd = board->getBoardConfiguration();
             if (qwd == NULL) {
                 qwd = new DefaultHwSettingsWidget(this, true);
-            } else {
+            }
+
+            if (!board->queryCapabilities(Core::IBoardType::BOARD_CAPABILITIES_OSD)) {
+                hasOSD = false;
```

Comment:
Can we do `if (index == ConfigGadgetWidget::osd) index = ConfigGadgetWidget::hardware;` here?